src/ceph-volume/ceph_volume/util/disk.py

```diff
@@ -170,6 +170,23 @@ def device_family(device):
     return devices
 
 
+def udevadm_property(device, properties=[]):
+    """
+    Query udevadm for information about device properties.
+    Optionally pass a list of properties to return. A requested property might
+    not be returned if not present.
+    """
+    cmd = ['udevadm', 'info', '--query=property', device]
```

Comment:
Something good to keep in mind for containers, this will only work if the container is privileged (`--privileged`).

Reply:
Actually not really, my container setup (highly privileged) returns only a few fields and not the one you're looking for:

```
[root@60a8b848d66e /]# udevadm info --query=property /dev/sdb
DEVNAME=/dev/sdb
DEVPATH=/devices/pci0000:00/0000:00:02.2/0000:02:00.0/host0/target0:1:0/0:1:0:1/block/sdb
DEVTYPE=disk
MAJOR=8
MINOR=16
SUBSYSTEM=block
```

Reply:
Ok answering my own question :)
We must bindmount `/var/run/udev/` to get what we want, see `-v /var/run/udev/:/var/run/udev/`

Reply:
Right. I kept the implementation close to what c++ land was doing. I think something based on ```/sys/block/$dev/``` would also work.

Also afaiu its yet unclear if and when we can run Ceph in unpreviliged containers.

Reply:
I don't think we can achieve non-privileged containers yet, but if we can retry the number of bindmount to apply by doing a "walkpath" in `/sys` approach that'd be easier for everyone using containers I believe :).

Reply:
ceph-volume has to require super user permissions in a ton of calls (look for `@needs_root` decorator in  class methods). Not only because LVM requires it, but also because there are a lot of utilities that need it. I doubt this can be simplified for only a subset of paths

Reply:
IMO we should rely on udev here.  Getting the same info ourselves out of /sys is complex and error-prone--that's why we are using udev in the first place.

Reply:
I have put in a fair bit of thought to running Ceph without privileged containers, and my thoughts have been that in the short-term, privileged containers are the best option for getting main functionality in place initially. Once things are working well, paring down privileges in containers are still extremely valuable optimizations to make.

Thoughts I have around OSDs in Rook are that the relatively short-lived `rook-ceph-osd-prepare-...` pods in rook are at less risk for security concerns due to privilege than are the constantly-running `rook-ceph-osd-<id>` pods. If `ceph-volume` can be limited to the prepare pod (which I believe is the intent), focusing effort on reducing permissions in the osd pods themselves is the more important place to focus on privilege needs. OSD daemons also likely represent the largest percentage of running Ceph daemons by far, so to my reasoning, security enhancements to the OSD daemons have the biggest return on investment for developer time spent.

A related writeup of my now probably outdated investigation as pertains to Rook is here: https://github.com/rook/rook/issues/1944